releng/release-go-changelog-update

```diff
@@ -114,6 +114,36 @@ def main(next_ver: str, today: datetime.date, quiet: bool=False, commit: bool =
 
         git_add("CHANGELOG.md")
 
+    with check(f"Updating releaseNotes.yml with date for {next_ver}..."):
+        state = 0
+
+        for line in fileinput.FileInput("docs/releaseNotes.yml", inplace=True):
+            if state == 0:
+                if line.strip() == f"- version: {next_ver}":
+                    state = 1
+            elif state == 1:
+                m = re.match(r"(\s+date:)\s+.*$", line)
+
+                if m:
+                    line = f"{m.group(1)} '{today.strftime('%Y-%m-%d')}'\n"
+                    state = 2
+            
+            sys.stdout.write(line)
+
+        if state != 2:
+            # Something didn't go right.
+            sadness = f"""
+ ==> ERROR: could not find the date for version '{next_ver}' in 
+     docs/releaseNotes.yml. Make sure that the block for that release
+     starts with
+
+     - version: {next_ver}
+       date: ....
+
+     as the start of the block."""
+
+            raise Exception(sadness)
+
```

Comment:
So why not delete the `CHANGELOG.md` part of this script, and instead have it run `make $PWD/CHANGELOG.md` after this `releaseNotes.yml` part?